http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -699,54 +698,77 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
 
       // Step 2: call LZ for LZ resources
       lzResourcesByPurpose <- wsmDao.listLandingZoneResourcesByType(landingZoneId, userToken)
+      groupedLzResources = lzResourcesByPurpose.foldMap(a =>
```

Comment:
It was bugging me that we iterated through the LZ resource each time we looked up a resource, so I grouped it to a `Map[(purpose, type), resource]` first.